eclipse/src/saros/ui/wizards/pages/EnterProjectNamePage.java

```diff
@@ -46,45 +47,57 @@
 /** A wizard page that allows to enter the new project name or to choose to overwrite a project. */
 public class EnterProjectNamePage extends WizardPage {
 
-  private static final Logger log = Logger.getLogger(EnterProjectNamePage.class.getName());
+  private static final Logger log = Logger.getLogger(EnterProjectNamePage.class);
 
   private final JID peer;
 
   private final ISarosSession session;
 
-  private final Map<String, String> remoteProjectMapping;
+  /** Map containing the mapping from the remote project id to the remote project name. */
+  private final Map<String, String> remoteProjectIdToNameMapping;
+
   private final Map<String, ProjectOptionComposite> projectOptionComposites =
       new HashMap<String, ProjectOptionComposite>();
 
   /** Map containing the current error messages for every project id. */
-  private Map<String, String> currentErrors = new HashMap<String, String>();
+  private final Map<String, String> currentErrors = new HashMap<>();
 
-  private IConnectionManager connectionManager;
+  /**
+   * Map containing the desired project name mapping as remote project name to local project name
+   */
+  private final Map<String, String> desiredRemoteToLocalProjectNameMapping;
 
-  private Preferences preferences;
+  private final IConnectionManager connectionManager;
 
-  private boolean flashState;
+  private final Preferences preferences;
 
   private final Set<String> unsupportedCharsets = new HashSet<String>();
 
+  private boolean flashState;
+
   public EnterProjectNamePage(
       ISarosSession session,
       IConnectionManager connectionManager,
       Preferences preferences,
       JID peer,
-      List<ProjectNegotiationData> projectNegotiationData) {
+      List<ProjectNegotiationData> projectNegotiationData,
+      Map<String, String> desiredRemoteToLocalProjectNameMapping) {
 
     super(Messages.EnterProjectNamePage_title);
     this.session = session;
     this.connectionManager = connectionManager;
     this.preferences = preferences;
+    this.desiredRemoteToLocalProjectNameMapping =
```

Comment:
Not strictly necessary as, currently, the passed mapping can't be `null` (`getLastProjectNameMapping(peer)` returns an empty hash map if no mapping is found).

But you can still keep it to protect against future 'misuse' if you would prefer. But I would move this to the end of the initialization block (behind `this.peer = peer;`) or avoid the inline-nullcheck to make this easier to read.

Reply:
Moved to the end to improve readability.